framework/src/packets/raw.rs

```diff
@@ -1,17 +1,45 @@
+use std::cmp;
 use common::Result;
 use native::zcsi::{mbuf_alloc, MBuf};
 use packets::{buffer, Header, Packet};
+use packets::buffer::read_slice;
 
 /// Unit header
 impl Header for () {}
 
 /// The raw network packet
 ///
 /// Simply a wrapper around the underlying buffer with packet semantic
+#[derive(Debug)]
 pub struct RawPacket {
     mbuf: *mut MBuf,
 }
 
+/// Compare RawPackets. This probably isn't something you want to be doing a lot of at runtime.
+impl PartialEq for RawPacket {
+    fn eq(&self, other: &RawPacket) -> bool {
+        unsafe {
+            if (*self.mbuf).data_len() != (*other.mbuf).data_len() {
+                return false;
+            }
+            let mut offset = 0;
+            let mut remaining = (*self.mbuf).data_len();
+            while remaining > 0 {
+                let to_read = cmp::min(remaining, 32);
```

Comment:
buffer read is zero copy. why not read the whole thing as a slice and compare instead of looping through 32 bytes a time?